extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/MethodNameParser.java

```diff
@@ -253,6 +255,74 @@ public Result parse(MethodInfo methodInfo) {
 
             String upperPrefix = (ignoreCase || allIgnoreCase) ? "UPPER(" : "";
             String upperSuffix = (ignoreCase || allIgnoreCase) ? ")" : "";
+
+            // If the fieldName is not a field in the class and in camelcase format,
+            // then split it as hierarchy of fields
+            if (entityClass.field(fieldName) == null) {
+                StringBuilder finalName = new StringBuilder(fieldName);
+
+                Set<String> childFields = new HashSet<>();
+
+                childFields.addAll(entityClass.fields().stream()
+                        .map(FieldInfo::name)
+                        .collect(Collectors.toList()));
+
+                // Collecting the current class fields
+                if (indexView.getClassByName(fieldInfo.type().name()) != null) {
+                    childFields.addAll(
+                            indexView.getClassByName(fieldInfo.type().name()).fields()
+                                    .stream()
+                                    .map(FieldInfo::name)
+                                    .collect(Collectors.toList()));
+                }
+
+                // Collecting the inherited fields from the superclass of the actual class
+                if (entityClass.superClassType().name() != null &&
```

Comment:
As it seems that `indexView.getClassByName((entityClass.superClassType().name())` and `indexView.getClassByName((fieldInfo.type().name())` are used a few times, best move them to a variable with a proper name